Add suport for TI INA238 power monitor

jsarrett opened the pull request:
This commit adds support for the TI INA238 power monitor IC via I2C.  It also builds the driver as a module by default.

I needed it on a custom CM4 board, and it would be much nicer if it were in-tree for the future.

jsarrett commented:
according to https://docs.kernel.org/devicetree/bindings/dts-coding-style.html Device tree files should be indented with Kernel style (i.e. spaces) why is checkpatch review complaining about that?

pelwell commented:
As long as indentation is consistent - single TABs, or either 4 or 8 spaces - we'll accept it. However, your overlay switches from TABs to spaces in places. Other than that, my problems with the overlay are:
1. I think it belongs in the i2c-sensor overlay. There you can benefit from the existing boiler plate for bus selection (which is about to change anyway).
2. It's not documented. As part of i2c-sensor you won't need to document all the i2c bus selection parameters.

jsarrett commented:
Will do!  Do you prefer a separate commit for the device tree and defconfig
change or all in one patch?

-James

On Fri, Jan 24, 2025 at 1:18 AM Phil Elwell ***@***.***>
wrote:

> As long as indentation is consistent - single TABs, or either 4 or 8
> spaces - we'll accept it. However, your overlay switches from TABs to
> spaces in places. Other than that, my problems with the overlay are:
>
>    1. I think it belongs in the i2c-sensor overlay. There you can benefit
>    from the existing boiler plate for bus selection (which is about to change
>    anyway).
>    2. It's not documented. As part of i2c-sensor you won't need to
>    document all the i2c bus selection parameters.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/raspberrypi/linux/pull/6629#issuecomment-2612040097>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZHTDDDXOOYPITSP6H26SL2MIAP7AVCNFSM6AAAAABVY6FJL2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMJSGA2DAMBZG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
---
"Never trust a man who can count to 1024 on his fingers."
---

pelwell commented:
Two commits - one for DTS, one for the defconfigs - in one PR is best.

jsarrett commented:
> Two commits - one for DTS, one for the defconfigs - in one PR is best.

Done!  LMK if you want more/different changes.  I've got another sensor to make a PR for as well, so the more I know how you guys want things the easier that will be.

pelwell commented:
The overlay check in the auto-builds is complaining that you've not described your additions in the overlays README: https://github.com/raspberrypi/linux/actions/runs/12961018541/job/36157211009?pr=6629#step:4:575

jsarrett commented:
So the devicetree parameters I added _are_ documented in Documentation/devicetree/bindings/hwmon/ti,ina2xx.yaml, as noted in Documentation/hwmon/ina238.rst.  Do we care that the dtoverlay checks script complains?  If so, how do I fix that/where is it's code?

jsarrett committed:
add ina238 to i2c-sensors

This patch adds the ina238 device tree parameters to the i2c-sensors
overlay.  The ina238 driver needs 2 configuration parameters, shut_resistor
and ti,shunt-gain in addition to it's address, so they are added as well.

jsarrett committed:
configs: add CONFIG_SENSORS_INA238=m
